src/pperm.c

```diff
@@ -6322,9 +6212,18 @@ Obj OnSetsPPerm(Obj set, Obj f)
                     *ptres++ = tmp;
                 }
             }
-            else { /* this case cannot occur since I think POW is not defined
-                      */
-                ErrorQuit("not yet implemented!", 0L, 0L);
+            else {
+                // This case does not work since PowIntPPerm2/4 only work for
+                // INTOBJ's (i.e. small integers). The analogous code in
+                // permutat.c uses the macro POW, which calls PowIntPerm2/4,
+                // which if called with a non-small positive integer returns
+                // that integer, since every permutation fixes every positive
+                // integer (small or not). The methods for PowIntPPerm2/4 give
+                // an error if the argument is a non-small integer, and so we
+                // cannot call POW here.
+                ErrorQuit("the first argument (a set) must consist of "
+                          "integers less than 2 ^ %d but contains %s",
+                          NR_SMALL_INT_BITS, (Int)TNAM_OBJ(*ptset));
```

Comment:
The standard way in the kernel to phrase such an error would be this: `ErrorQuit("<set> must be a list of small integers", 0,0);`. This uses that this kernel function is called for `OnSets`, which is documented as `OnSets( set, g )`.

But of course the `OnSets` function for permutation does not contain such a message -- because it relies on `POW` producing an error, if needed. And that in turn makes it possible for the user resp. packages to install custom methods for `\^` -- which then in turn would automatically be useable via `OnSets`.

I am not saying that's necessarily a good idea, but it's a possibility there. We could of course make it stricter, but then it would be kinda nice if the pperm and perm code would behave similarly for this.

(That said, I'd accept either approach in this PR, we can work on unification in another PR)

Reply:
I'll update this.

Reply:
I agree it would be better if the pperm code worked in more or less the same way as the permutat code, but didn't want to make further changes in this PR.